app/controllers/application_controller.rb

```diff
@@ -2,4 +2,9 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_filter :set_app_info_headers
+
+  def set_app_info_headers
+  	headers['X-GitSHA'] = AppInfo::GIT_REVISION
```

Comment:
as reference (since we are all learning), here are headers from GitHub's API:
```curl -i https://api.github.com/users/shauni```
HTTP/1.1 200 OK
Server: GitHub.com
Date: Wed, 13 Jul 2016 23:00:48 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 1231
Status: 200 OK
X-RateLimit-Limit: 60
X-RateLimit-Remaining: 59
X-RateLimit-Reset: 1468454448
Cache-Control: public, max-age=60, s-maxage=60
Vary: Accept
ETag: "b7bd0fbf91d786c47fbef0e740ce05f3"
Last-Modified: Tue, 31 May 2016 15:29:39 GMT
X-GitHub-Media-Type: github.v3
Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
Access-Control-Allow-Origin: *
Content-Security-Policy: default-src 'none'
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
Vary: Accept-Encoding
X-Served-By: d594a23ec74671eba905bf91ef329026
X-GitHub-Request-Id: 4C77E9CC:7ABD:1144F948:5786C820